x/delegation/keeper/delegation.go

```diff
@@ -63,10 +65,20 @@ func (k *Keeper) delegateTo(
 		return err
 	}
 
-	err = k.assetsKeeper.UpdateOperatorAssetState(ctx, params.OperatorAddress, assetID, assetstype.DeltaOperatorSingleAsset{
+	deltaOperatorAsset := assetstype.DeltaOperatorSingleAsset{
 		TotalAmount: params.OpAmount,
 		TotalShare:  share,
-	})
+	}
+	// Check if the staker belongs to the delegated operator. Increase the operator's share if yes.
+	operator, err := k.GetSelfDelegatedOperator(ctx, stakerID)
+	if err != nil {
+		return err
+	}
+	if operator != "" && operator == params.OperatorAddress.String() {
```

Comment:
Since `params.OperatorAddress.String()` is not `""`, the first check is unnecessary.
```suggestion
	if operator == params.OperatorAddress.String() {
```

Reply:
Thanks